client/components/Forms/EventForm.js

```diff
@@ -344,7 +346,7 @@ class EventForm extends Component {
     const categoryClass = categoryIsValid ? 'category-label' : 'category-label category-error';
     const timezoneClass = timezoneIsValid ? 'category-label' : 'category-label category-error';
     const categoryTitle = category || 'Select Category';
-    const eventChannel = slackChannel || 'Select Slack Channel';
+    const eventChannel = slackChannel ?  slackChannels.filter(channel => channel.id === slackChannel)[0].name : 'Select Slack Channel';
```

Comment:
Although this takes more lines of code, I think it's safe to also check if the slack channel was found. For example, if for any reason the slack channel was deleted from slack, and a user tries to edit an event that used to have that slack channel this will throw a TypeError.
```
Uncaught TypeError: Cannot read property `name` of undefined
```

```suggestion
   let eventChannel = 'Select Slack Channel';
    if (slackChannel) {
      const [foundChannel] = slackChannels.filter(channel => channel.id === slackChannel);
      eventChannel = !foundChannel ? eventChannel : foundChannel.name;
    }
```

Kindly help look into that,